SSPSolution/GraphicsDLL/DeferredShader.cpp

```diff
@@ -501,6 +502,70 @@ int DeferredShader::Draw(Resources::Model * model)
 	return 0;
 }
 
+int DeferredShader::Draw(Resources::Model * model, GraphicsComponent * component)
+{
+	ConstantBufferHandler::GetInstance()->world.UpdateBuffer(&component->worldMatrix);
+	Resources::Mesh* meshPtr = model->GetMesh();
+	ID3D11Buffer* vBuf		 = meshPtr->GetVerticesBuffer();
+	ID3D11Buffer* iBuf = meshPtr->GetIndicesBuffer();
+	UINT32 offset = 0;
+	this->m_deviceContext->IASetVertexBuffers(0, 1, &vBuf, &m_vertexSize, &offset);
+	this->m_deviceContext->IASetIndexBuffer(iBuf, DXGI_FORMAT::DXGI_FORMAT_R32_UINT, 0);
+
+	Resources::Material * mat = model->GetMaterial();
+	Resources::Texture** textures = mat->GetAllTextures();
+	ID3D11ShaderResourceView* resViews[5];
+	UINT numViews = 0;
+	for (size_t i = 0; i < 5; i++)
+	{
+		if (textures[i] == nullptr)
+			continue;
+
+		resViews[numViews] = textures[i]->GetResourceView();
+		numViews += 1;
+	}
+
+
+	this->m_deviceContext->PSSetShaderResources(0, numViews, resViews);
+
+	this->m_deviceContext->DrawIndexed(meshPtr->GetNumIndices(), 0, 0);
+
+
+	return 0;
+}
+
+int DeferredShader::Draw(Resources::Model * model, penis * component)
```

Comment:
Well now. Let us not forget this temporary pointer for our future release build.